Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/SortAdapter.cpp

```diff
@@ -60,16 +115,58 @@ namespace AZ::DocumentPropertyEditor
         {
             GenerateFullTree();
         }
-        NotifyResetDocument(DocumentResetType::HardReset);
+        NotifyResetDocument();
     }
 
-    void RowSortAdapter::HandleDomChange([[maybe_unused]] const Dom::Patch& patch)
+    void RowSortAdapter::HandleDomChange(const Dom::Patch& patch)
     {
         if (m_sortActive)
         {
-            // TODO: handle changes upstream by making  custom patches. This is not currently
-            // possible, because a sort change is mostly move operations, and the DPE doesn't handle moves yet.
-            HandleReset();
+            bool needsReset = false;
+            //const auto& sourceContents = m_sourceAdapter->GetContents();
+
+            Dom::Patch outgoingPatch;
+            for (auto operationIterator = patch.begin(), endIterator = patch.end(); !needsReset && operationIterator != endIterator;
+                 ++operationIterator)
```

Comment:
suggestion: Any reason why we can't use a range-based for loop here? I see we have the additional check, but we could check `needsReset` at the start of the loop potentially? Not a huge deal it's just iterators do usually imply something more involved is going on like elements being inserted/removed